File: Emulator/src/main/content/global/skill/summoning/familiar/dialogue/HoneyBadgerDialogue.java
package content.global.skill.summoning.familiar.dialogue;

import core.game.dialogue.Dialogue;
import core.game.dialogue.FaceAnim;
import core.game.node.entity.npc.NPC;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;
import org.rs.consts.NPCs;

import static core.tools.DialogueHelperKt.END_DIALOGUE;

/**
 * The type Honey badger dialogue.
 */
@Initializable
public class HoneyBadgerDialogue extends Dialogue {

    @Override
    public Dialogue newInstance(Player player) {
        return new HoneyBadgerDialogue(player);
    }

    /**
     * Instantiates a new Honey badger dialogue.
     */
    public HoneyBadgerDialogue() {}

    /**
     * Instantiates a new Honey badger dialogue.
     *
     * @param player the player
     */
    public HoneyBadgerDialogue(Player player) {
        super(player);
    }

    @Override
    public boolean open(Object... args) {
        npc = (NPC) args[0];
        switch ((int) (Math.random() * 5)) {
            case 0:
                npcl(FaceAnim.CHILD_NORMAL, "An outpouring of sanity-straining abuse*");
                stage = 0;
                break;
            case 1:
                npcl(FaceAnim.CHILD_NORMAL, "An outpouring of spittal-flecked insults.*");
                stage = 0;
                break;
            case 2:
                npcl(FaceAnim.CHILD_NORMAL, "A lambasting of visibly illustrated obscenities.*");
                stage = 0;
                break;
            case 3:
                npcl(FaceAnim.CHILD_NORMAL, "A tirade of biologically questionable threats*");
                stage = 0;
                break;
            case 4:
                npcl(FaceAnim.CHILD_NORMAL, "A stream of eye-watering crudities*");
                stage = 0;
                break;
        }
        return true;
    }

    @Override
    public boolean handle(int interfaceId, int buttonId) {
        if (stage == 0) {
            playerl(FaceAnim.FRIENDLY, "Why do I talk to you again?");
            stage = END_DIALOGUE;
        }
        return true;
    }

    @Override
    public int[] getIds() {
        return new int[]{NPCs.HONEY_BADGER_6845, NPCs.HONEY_BADGER_6846};
    }
}

File: Emulator/src/main/content/global/skill/summoning/familiar/dialogue/HoneyBadgerDialogue.kt
package content.global.skill.summoning.familiar.dialogue

import core.game.dialogue.Dialogue
import core.game.dialogue.FaceAnim
import core.game.node.entity.npc.NPC
import core.game.node.entity.player.Player
import core.plugin.Initializable
import core.tools.END_DIALOGUE
import shared.consts.NPCs

/**
 * The type Honey badger dialogue.
 */
@Initializable
class HoneyBadgerDialogue : Dialogue {
    override fun newInstance(player: Player?): Dialogue = HoneyBadgerDialogue(player)

    /**
     * Instantiates a new Honey badger dialogue.
     */
    constructor()

    /**
     * Instantiates a new Honey badger dialogue.
     *
     * @param player the player
     */
    constructor(player: Player?) : super(player)

    override fun open(vararg args: Any?): Boolean {
        npc = args[0] as NPC
        when ((Math.random() * 5).toInt()) {
            0 -> {
                npcl(FaceAnim.CHILD_NORMAL, "An outpouring of sanity-straining abuse*")
                stage = 0
            }

            1 -> {
                npcl(FaceAnim.CHILD_NORMAL, "An outpouring of spittal-flecked insults.*")
                stage = 0
            }

            2 -> {
                npcl(FaceAnim.CHILD_NORMAL, "A lambasting of visibly illustrated obscenities.*")
                stage = 0
            }

            3 -> {
                npcl(FaceAnim.CHILD_NORMAL, "A tirade of biologically questionable threats*")
                stage = 0
            }

            4 -> {
                npcl(FaceAnim.CHILD_NORMAL, "A stream of eye-watering crudities*")
                stage = 0
            }
        }
        return true
    }

    override fun handle(
        interfaceId: Int,
        buttonId: Int,
    ): Boolean {
        if (stage == 0) {
            playerl(FaceAnim.FRIENDLY, "Why do I talk to you again?")
            stage = END_DIALOGUE
        }
        return true
    }

    override fun getIds(): IntArray = intArrayOf(NPCs.HONEY_BADGER_6845, NPCs.HONEY_BADGER_6846)
}
